api/client/client_test.go

```diff
@@ -136,6 +137,38 @@ func TestClient_NewCollection(t *testing.T) {
 	})
 }
 
+func TestIt(t *testing.T) {
+	type Owner struct {
+		ID       core.InstanceID `json:"_id"`
+		ThreadID thread.ID       `json:"threadId"`
+	}
+	collectionName := "owners"
+
+	client, done := setup(t)
+	defer done()
+
+	id := thread.NewIDV1(thread.Raw, 32)
+	err := client.NewDB(context.Background(), id)
+	checkErr(t, err)
+	sch := util.SchemaFromInstance(Owner{}, false)
+	schBytes, err := json.MarshalIndent(sch, "", "  ")
+	checkErr(t, err)
+	schStr := string(schBytes)
+	t.Logf("\n%v", schStr)
+	err = client.NewCollection(context.Background(), id, db.CollectionConfig{Name: collectionName, Schema: sch})
+	checkErr(t, err)
+
+	owner := &Owner{
+		ID:       "",
+		ThreadID: id,
+	}
+
+	_, err = client.Create(context.Background(), id, collectionName, Instances{owner})
+	if err != nil {
+		t.Fatalf("failed to create instance: %v", err)
```

Comment:
When you try to save a `Owner` here, you get the error:
```
failed to create instance: rpc error: code = Unknown desc = instance doesn't correspond to schema
```

Reply:
This is a bit mysterious, but it seems has something to do with the nested private `str string` inside `thread.ID` and the way that interacts with json schema and/or json marshaling/unmarshaling